src/App/integrationTests/app.userDeniedSyncPullNullLastRevisionNumbers.test.js

```diff
@@ -0,0 +1,228 @@
+          error: {
+            code: 403,
+            record_ids: [],
+          },
+        },
+        project_profiles: {
+          last_revision_num: 'server',
+          error: {
+            code: 403,
+            record_ids: [],
+          },
+        },
+        project_sites: {
+          last_revision_num: 'server',
+          error: {
+            code: 403,
+            record_ids: [],
+          },
+        },
+      }
+
+      return res.once(ctx.json(responseWithSyncErrors))
+    }),
+  )
+
+  const projectCardForProjectWithId1 = screen.getAllByRole('listitem')[0]
+  const linkToCollectingPageForProjectWithId1 = within(projectCardForProjectWithId1).getByRole(
+    'link',
+    { name: 'Collect' },
+  )
+
+  userEvent.click(linkToCollectingPageForProjectWithId1)
+
+  await screen.findByLabelText('project pages loading indicator')
+  await waitForElementToBeRemoved(() => screen.queryByLabelText('project pages loading indicator'))
+
+  const lastRevisionProject1BenthicAttributesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'benthic_attributes' })
+      .toArray()
+
+  const lastRevisionProject1ChoicesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'choices' })
+      .toArray()
+  const lastRevisionProject1FishFamiliesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'fish_families' })
+      .toArray()
+  const lastRevisionProject1FishGeneraAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'fish_genera' })
+      .toArray()
+  const lastRevisionProject1FishSpeciesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'fish_species' })
+      .toArray()
+
+  const lastRevisionProject1CollectRecordsAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'collect_records', projectId: '1' })
+      .toArray()
+
+  const lastRevisionProject1ManagementsAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'project_managements', projectId: '1' })
+      .toArray()
+
+  const lastRevisionProject1ProfilesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'project_profiles', projectId: '1' })
+      .toArray()
+
+  const lastRevisionProject1SitesAfterSyncError =
+    await dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled
+      .where({ dataType: 'project_sites', projectId: '1' })
+      .toArray()
+
+  // when a user is denied a sync pull, we want to not mess with the last revision numbers for untouched 'tables'
+  expect(lastRevisionProject1BenthicAttributesAfterSyncError[0].lastRevisionNumber).toEqual(
+    'server',
```

Comment:
why does the` lastRevisionNumber `equal 'server' ?

Reply:
We're testing that these types of data dont have their last revision numbers wiped. the mock api above sets the last revision numbers to server, so we want to see that in the browser, we have kept whatever the server has sent.

For the other tables below, we want to have wiped last revision number, so that next time we pull, we get _everything_, not just what ever has changed on the api. Last revision number is for the api to determine how uptodate/ in sync the browser is and it only sends changes. Usually the front end doesnt touch last revision numbers, it just stores whatever the api sends, except here where it wipes last revision numbers after we wipe the data corresponding to the last revision numbers.

Reply:
ah ok that makes sense - we just want to ensure something is there. thanks for explaining!